### DISCUSSION TITLE
force:org:list --json not showing namespace of developer edition

### DISCUSSION THREAD

**Discussion by ragoergens** [forcedotcom/cli#2351] [Ideas]
### Summary

The namespace or namespacePrefix attribute of a developer edition org having a namespace is not shown in sfdx.

### Steps To Reproduce:

1.  Create developer edition
1.  Define namespace
1.  Authorise developer edition using force:web:login
1.  Do force:org:list --json 

### Expected result

The namespace or namespacePrefix attribute is shown.

### Actual result

The namespace or namespacePrefix attribute is not shown.

{
  "status": 0,
  "result": {
    "nonScratchOrgs": [
      {
        "accessToken": "[redacted]",
        "instanceUrl": "https://[redacted]-dev-ed.my.salesforce.com",
        "orgId": "[redacted]",
        "username": "[redacted]",
        "loginUrl": "https://login.salesforce.com/",
        "clientId": "PlatformCLI",
        "isDevHub": false,
        "instanceApiVersion": "56.0",
        "instanceApiVersionLastRetrieved": "03/11/2022, 21:51:55",
        "alias": "[redacted]",
        "lastUsed": "2022-11-03T21:23:22.394Z",
        "connectedStatus": "Connected"
      }
    ],
    "scratchOrgs": []
  }
}

### System Information

{
  "cliVersion": "sfdx-cli/7.174.0",
  "architecture": "win32-x64",
  "nodeVersion": "node-v16.17.1",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 1.3.1 (core)",
    "@oclif/plugin-commands 2.2.0 (core)",
    "@oclif/plugin-help 5.1.14 (core)",
    "@oclif/plugin-not-found 2.3.3 (core)",
    "@oclif/plugin-plugins 2.1.1 (core)",
    "@oclif/plugin-update 3.0.2 (core)",
    "@oclif/plugin-version 1.1.2 (core)",
    "@oclif/plugin-warn-if-update-available 2.0.6 (core)",
    "@oclif/plugin-which 2.1.0 (core)",
    "alias 2.1.5 (core)",
    "apex 1.3.0 (core)",
    "auth 2.2.15 (core)",
    "community 2.0.10 (core)",
    "config 1.4.21 (core)",
    "custom-metadata 2.0.2 (core)",
    "data 2.1.4 (core)",
    "generator 2.0.7 (core)",
    "info 2.2.0 (core)",
    "limits 2.0.5 (core)",
    "org 2.2.9 (core)",
    "packaging 1.9.4 (core)",
    "schema 2.1.10 (core)",
    "signups 1.2.8 (core)",
    "source 2.0.25 (core)",
    "telemetry 2.0.2 (core)",
    "templates 55.1.0 (core)",
    "trust 2.0.4 (core)",
    "user 2.1.13 (core)",
    "@salesforce/sfdx-plugin-lwc-test 1.0.1 (core)",
    "salesforce-alm 54.8.3 (core)"
  ],
  "osVersion": "Windows_NT 10.0.19044",
  "shell": "cmd.exe",
  "rootPath": "C:\\Users\\[redacted]\\AppData\\Local\\sfdx\\client\\7.174.0-a2f8d31"
}

[a comment by an automated account is not shown]

**Comment by mshanemc** [comment]
we store the namespace when scratch orgs get created since we know it.  We've never asked an org during auth about its namespace, but we certainly could (and then store that so it shows up for list/display).

This would also help the situation where you auth to a previously created scratch org with a namespace (those would also show up as blank).

**Comment by ragoergens** [comment]
Sounds great, thanks! This kind of consistency would be helpful.

**Comment by cristiand391** [comment]
We might add this in the future but I'll mark this as `help wanted` in case you or anyone else want to add this feature.

As Shane mentioned, we already store the namespace prefix when a scratch org is created:
https://github.com/forcedotcom/sfdx-core/blob/main/src/org/authInfo.ts#L88

(you can confirm this if you linked a namespace to your hub and created a scratch org, run `force:org:list` to get the username and then look at the auth file in `~/.sfdx/<USERNAME>.json`, you should see the `namespacePrefix` field in the json).

### `force:org:list`: add `NAMESPACE PREFIX` column
this should be pretty easy, just add an object with the header for the table and the getter for the `namespacePrefix` field (you might need to add the field to the `ExtendedAuthFields` type too). the json output will include the new field too, no need to add anything else there.

see how the `ALIAS` column work, you can add the namespace prefix column next to it:
https://github.com/salesforcecli/plugin-org/blob/441f4f38e85e32bd11b4440e6c24aad258604fcc/src/commands/org/list.ts#L183-L186

feel free to post here any questions you have.

**Comment by ragoergens** [comment]
Thanks for this! I'm not yet ts literate, but I'll try to ask a colleague who is.

**Comment by atsutton** [comment]
@cristiand391 - I would like to work on this. Is that alright?

**Comment by cristiand391** [comment]
Hi @atsutton 👋🏼 
Sure! let me know if you have any question, see also our contribution guide: https://github.com/forcedotcom/cli/blob/main/.github/CONTRIBUTING.md

**Comment by atsutton** [comment]
@cristiand391 - Pull request is ready for review. Please let me know if you need anything else. 
https://github.com/salesforcecli/plugin-org/pull/596

**Comment by mshanemc** [comment]
this was delivered in the linked PR and the PR linked from there
